# Computational Astrophysics
## Exercises 02.

**Name:** 

---

## 1. Imagen de NGC1300

We will use a .fits image of galaxy NGC1300 obtained from the SIMBAD Database. Go to

http://cdsportal.u-strasbg.fr/?target=NGC1300

In the Tabular Data section, click on the VizieR images tab and download the images corresponding to the filters

B (Effective Wavelength midpoint: $445$ nm [blue])

I (Effective Wavelength midpoint: $806$ nm [near IR])

K (Effective Wavelength midpoint: $2190$ nm [near IR])

Obtain a rgb-compound image of the galaxy. Try different values for the parameters *Q* and *stretch* in the `make_lupton_rgb` function until you obtain a good looking image.

---

## 2. Color-Color Diagram of a Stars sample
**`Carroll and Ostlie (2017). Section 3.6`**

1. Use SkyServer's Object Explorer

https://skyserver.sdss.org/dr1/en/tools/explore/obj.asp?ra=179.88431&dec=-0.282125

to find the stars with the following object IDs in the SDSS database. Look at the object's magnitudes in SDSS's five filters - they are the quantities u, g, r, i, and z in the second row next to the object's image. 

| Object ID  | RA  | Dec  |
| :---: | :---: | :---: |
| 582093483279581263 | 179.88431 | -0.28125|
| 582093483279581296 | 179.94806 | -0.2724 |
| 582093482742710367 | 179.97106 | -0.82002|
| 582093499934703623 | 179.8339  | -0.55558|
| 582093499934703856 | 179.89941 | -0.61227|
| 582093499934703933 | 179.97446 | -0.55081|
| 582093499934703813 | 179.85499 | -0.4708 |
| 582093499934703876 | 179.91213 | -0.50101|
| 582093499934703941 | 179.97887 |-0.57861 |
| 582093483279515723 | 179.789   | -0.29908|
| 582093499934834837 | 180.20067 |-0.57081 |
| 582093483279646824 | 180.10892 | -0.34719|
| 582093482742775943 | 180.11286 |-0.77366 |
| 582093482742775905 | 180.10103 | -0.64383|
| 582093482742710369 | 179.97602 |-0.66007 |
| 582093499934834834 | 180.19612 |-0.47336 |
| 582093482742775953 | 180.16034 |-0.72755 |

2. Save the data for the stars including the object ID and the five magnitudes of each star as a .txt and as a .csv files.

3. Find the *g-r*, *r-i*, *u-g* and *i-z* colors of each of the 17 stars.

4. Make a color-color diagram of the 17 stars, with *g-r* on the x-axis and *r-i* on the y-axis.

5. Which end of the line in your graph corresponds to hotter stars? Which corresponds to cooler stars? How do you know?

**Hint:** Your graph will show that hotter stars tend to follow the trend of a straight line, but cooler stars diverge from this trend. This means that hotter stars can be thought of as thermal sources, but cooler stars can not.

6. Make another color-color diagram of the 17 stars, with *u-g* on the x-axis and *g-r* on the y-axis. Once again, the hottest stars follow a linear trend, meaning they can be thought of as thermal sources. But in the *u-g/g-r* diagram, where does this trend begin to break down? What is the significance of this observation for thinking about real stars as thermal sources?

7. If you know about stellar evolution you can answer this question. What is the significance of the flat line at the top of the *u-g/g-r* diagram? What types of stars are these?

**Hint:** What does it mean for *g-r* to be constant as *u-g* changes?



---

## 3. Color-color Diagram of a Galaxies sample
**`Carroll and Ostlie (2017). Section 4.2`**<br>
**`Ivezic et. al. (2014). Section 1.5.3`**

In this exercise we will retrive a sampe of approximately 330000 galaxies from the SDSS database using the package `astroML`. Instead of entering the SDSS database directly we will use the dataset incorporated in the `astroML` package, which can be accesed 

In [4]:
from astroML.datasets import fetch_imaging_sample
data = fetch_imaging_sample()

**Important Note: The first time that you call this command, the code will send a http request and download the data from the web and this may be time consuming! On subsequent calls, the data will be loaded from the local disk.**

The command above stored the information in the array `data`. The shape of this array is

In [6]:
data.shape

(330753,)

This menas that we have 330753 objects in the dataset. The names of the stored attributes of these objects can be accesed via the `dtype` attribute,

In [8]:
data.dtype.names

('ra',
 'dec',
 'run',
 'rExtSFD',
 'uRaw',
 'gRaw',
 'rRaw',
 'iRaw',
 'zRaw',
 'uErr',
 'gErr',
 'rErr',
 'iErr',
 'zErr',
 'uRawPSF',
 'gRawPSF',
 'rRawPSF',
 'iRawPSF',
 'zRawPSF',
 'upsfErr',
 'gpsfErr',
 'rpsfErr',
 'ipsfErr',
 'zpsfErr',
 'type',
 'ISOLATED')

These features include the coordinates

'ra': right ascnesion\
'dec': declination\
'rExtSFD': r band extinsion from SFD\
'uRaw': raw photometric data in the u band\
'gRaw': raw photometric data in the g band\
'rRaw': raw photometric data in the r band\
'iRaw': raw photometric data in the i band\
'zRaw': raw photometric data in the z band\
'uErr': error data in the u band\
...

etc.

This dataset includes objects of the `'type' = 3` (mostly galaxies) and `'type' = 6` (mostly stars and quasars) according to the classification of the SDSS database.

Data from each feature can be easily accesed. For example, the right ascension of the first 10 samples are

In [9]:
data['ra'][:10]

array([0.358174, 0.358382, 0.357898, 0.35791 , 0.358881, 0.162688,
       0.142353, 0.142009, 0.121992, 0.221928])

We can extract some of the data from the loaded dataset. For example, lets extract the first 5000 objects of the type 6 (stars and quasars),

In [10]:
stars = data[data['type'] == 6][:5000]
stars.shape

(5000,)

In [11]:
stars.dtype.names

('ra',
 'dec',
 'run',
 'rExtSFD',
 'uRaw',
 'gRaw',
 'rRaw',
 'iRaw',
 'zRaw',
 'uErr',
 'gErr',
 'rErr',
 'iErr',
 'zErr',
 'uRawPSF',
 'gRawPSF',
 'rRawPSF',
 'iRawPSF',
 'zRawPSF',
 'upsfErr',
 'gpsfErr',
 'rpsfErr',
 'ipsfErr',
 'zpsfErr',
 'type',
 'ISOLATED')

1. Create a color-magnitude diagram for the first 5000 stars in the dataset, plotting the $g-r$ color in the x-axis vs. the $r$ magnitude in the y-axis.

2. Create a color-color diagram for the first 5000 stars in the dataset, plotting the $g-r$ color in the x-axis vs. the $r-i$ color in the y-axis.

3. Create a color-magnitude diagram for the first 5000 galaxies in the dataset, plotting the $g-r$ color in the x-axis vs. the $r$ magnitude in the y-axis.

4. Create a color-color diagram for the first 5000 galaxies in the dataset, plotting the $g-r$ color in the x-axis vs. the $r-i$ color in the y-axis.

**Hint:** For a better visualization, it is useful to invert the magnitude axis.

---
## 4. Spectroscopic Binary Data
**`Griffin, R.F. Obs. 130, p.358 (2010). Table VI`**
https://ui.adsabs.harvard.edu/abs/2010Obs...130..349G/abstract


Spectroscopic binary data both single line and double line was published in the journal "The Observatory" by R.F. Griffin as a long series of papers, describing the continuous work by Yoss and Griffin (1997, JA and A., 18,161). 

The observations have been obtained with photoelectric spectrum scanners and many of very long period have been published recently. 

The paper No. 211, Griffin, R.F. Obs. 130, p.358 (2010), includes data for the stars HD128642, HD144601, HD150172, and HD155641. In particular, the information in the Table VI, which reports the radial velocities of HD108613 recorded from 1973 to 2010 is given in the file HD108613.csv (it includes the Julian Date and radial velocity in km/s). No spectrum type is available and there is evidence of photometric variability of unknown origin. 

1. Use the data from file HD108613.csv to plot the radial velocity as function of the Julian date. Can you estimate the period of motion of this system?


¡¡ Happy Coding !!